insights-optimizing-octopus-webinar (#506)
OctopusDeploy/blog: blog/2021-08/insights-from-our-optimizing-octopus-webinar/index.md

---
title: Insights from our Optimizing Octopus webinar
description: Learn how to keep Octopus running smoothly and easily see what’s happening, with a little planning, maintenance and smart use of Octopus features.
author: andrew.corrigan@octopus.com
visibility: private
published: 2021-08-09-1400
metaImage: 
bannerImage: 
bannerImageAlt:
isFeatured: false
tags:
  - DevOps
---

> In [our recent webinar](https://octopus.com/events/optimizing-octopus-deploy-with-clear-measure), Clear Measure’s Chris Thomas joined Derek Campbell to talk about ways to optimize Octopus Deploy for the best possible experience.

Reviewer: Because the webinar is embedded a couple of paragraphs down, I don't think we need to include the link here and risk people clicking away.

Reviewer 2: Agree with this. Was from before I added the embed link.

> 

Reviewer: When I read: "See how to best organize Octopus instead", I expected a link to docs. To make it clear they're jumping ahead in the post, I think we could phrase it: Skip ahead to how best organize Octopus.

Reviewer 2: Agree with this, especially as a lot of my links do go to the documentation. Maybe 'Skip ahead **for** how to best organize Octopus' has nicer ring to it?

### Check SQL reports for problems

> First, we recommend running a ‘Index Physical Statistics’ report in SSMS to look for problems. 
> 
> To run the report, right-click your database from the list and select **{{Reports > Standard Reports > Index Physical Statistics}}**.

Reviewer: Moved the tool from the first sentence to the second. My thinking here is it provides a bit more context, SSMS > right click the database...
```suggestion
First, we recommend running a ‘Index Physical Statistics’ report to look for problems. 

To run the report, in SSMS right-click your database from the list and select **{{Reports > Standard Reports > Index Physical Statistics}}**.
```

Reviewer 2: Disagree with this one as we need it to be clear that they can only do this report in SSMS (other tools might have it or similar, I can't be sure).

The lead-in text after the H2 also suggests SSMS as a tool, rather the saying they should definitely use that.

Set the plan to run on a schedule that suits your business and select the following maintenance tasks:

- Check Database Integrity
- Reorganize Index
- Rebuild Index
- Update Statistics

Use others if needed, though we advise against shrinking the database for performance reasons.

As you progress through the wizard, you may want to change settings for each chosen task. We find the defaults are enough, but you can change things as you need. 

Run the maintenance plan once you’re happy.

If you prefer using scripts, try [Ola Hallengren’s SQL Server maintenance solution](https://ola.hallengren.com/), which achieves everything we outlined above.

## Organize Octopus

> 

Reviewer: Good practice feels a bit vague. Could we say something like:

Here are a few suggestions to keep your dashboard lean and easy to navigate.

Reviewer 2: Fine with this one.

We advise using fewer than 10 environments to reduce:

- Performance problems
- Project redundancy and overlap
- Unnecessary clutter on the dashboard

![Octopus dashboard with ideal environment setup, showing development pipeline that includes Dev, QA, Stage, UAT and Production](environments.png "width=500")

We find some customers use more environments than needed to manage lots of projects or deployment targets. Octopus includes better options, such as Project groups and Tenants.

### Group your projects

If you have many Projects, you can group them to help reduce visual noise on your dashboard.

![Octopus dashboard showing projects in Octopus, all grouped logically using Project Groups](projects.png "width=500")

To create a project group:

1.	Click **Projects** in Octopus’s top menu.

> 1.	Click **ADD GROUP** in the top right.
> 1.	Give your group a name and description, then click **SAVE**.

Reviewer: ```suggestion
1. Click **ADD GROUP** in the top right.
1. Give your group a name and description, then click **SAVE**.
```

To add an existing project to a group:

1.	Click **Projects** in Octopus’s top menu.
1.	Click on the project you need to move to the new group.
1.	Click **Settings** at the bottom of the left menu.
1.	Click **Project Group**, then select the destination group from the **Project group** dropdown menu.

To add a new project to a group:

1.	Click **Projects** in Octopus’s top menu.
1.	Click **ADD PROJECT** in the top right.
1.	Give your new project a name and click **SHOW ADVANCED**.
1.	Select a group from the **Project group** dropdown menu.
1.	Click **SAVE** to finish creating your new project.

### Use Tenants

> We designed Tenants to help customers who deliver Software as a Service (SaaS), but they’re also useful to organize things like:

Reviewer: Do we have thoughts on whether or not tenants should be capitalized? This feels like a tricky one, but I think tenants are common nouns. If we want to rephrase it slightly we could say, We design the Multi-tenancy feature to help...

Reviewer 2: I'm copying the UI here for clarity as we don't call it 'Multi-tenancy' in the app itself, and I think I've only used where it would need proper-nouning.

Reviewer: I think for this one, when we refer to the feature we should use the full name of the feature, Mult-tenancy, but tenants fits in some cases too.

Author: @leecarlon so I'm sure, do I just change the heading to Multi-tenancy, and then also the first sentence to say "We designed Multi-tenancy to help customers..."?
(The rest of the references are correct I believe.)

-	Geographical regions or data centers
-	Developers, testers or teams
-	Feature branches

![A set of tenants in Octopus, showing how they can group and manage deployment targets and demographics](tenants.png "width=500")

Check out our [documentation on Tenants](https://octopus.com/docs/tenants) for more information. 

Also, watch our follow-up webinar with Adam Close and Mark Harrison, [Better multi-tenancy deployments using Octopus Deploy](https://octopus.com/events/better-multi-tenancy-deployments-using-octopus-deploy).

### Delete your old releases and builds

It’s great practice to delete anything you no longer need from Octopus. Octopus Server is set to keep everything by default (Octopus Cloud defaults to 30 days), but you can use Lifecycle retention policies to auto-prune old releases and builds.

To check your retention policies:

1. Click **Library** in Octopus’s top menu.
1. Click **Lifecycles** from the left menu.

Here you can change retention policies for existing and default Lifecycles, or create new ones.

> By setting retention policies, files will disappear after the number of days or releases you set, saving you valuable disk space.

Reviewer: ```suggestion
By setting retention policies, files will be deleted after the number of days or releases you set, saving you valuable disk space.
```

Reviewer 2: Wanted to vito this change on passive language reasons, but I'm happy for a 'will delete' as a compromise?

Reviewer: Let's say, Octopus will delete the files after....
